include/qinterface.hpp

```diff
@@ -1869,6 +1846,22 @@ class QInterface {
      */
     virtual void ProbMaskAll(const bitCapInt& mask, real1* probsArray);
 
+    /**
+     * Statistical measure of masked permutation probability
+     *
+     * "qPowers" contains powers of 2^n representing bit n, to mask them in results in ascending order from qPowers[0]
+     * to qPowers[qPowerCount - 1]. "shots" specifies the number of samples to take as if totally re-preparing the
+     * pre-measurement state. This method returns a dictionary with key, (masked-order) measurement result, and value,
+     * number of "shots" which produced that measurement. This method does not "collapse" the state. (The idea is to
+     * efficiently simulate a potentially statistically random multiple re-preparations of the state right before
+     * measurement, and collect random measurement resutls, without forcing the user to re-prepare or "clone" the
```

Comment:
Spell check these comments.

Reply:
The English-major-in-a-past-life in me wrote this. I think it's grammatically and typographically correct, but I'll consider editing it for a less highfalutin construction, to favor clarity.